src/plugins/intel_cpu/src/transformations/cpu_opset/common/pass/markup_rope_inputs.hpp

```diff
@@ -0,0 +1,22 @@
+// Copyright (C) 2018-2024 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#pragma once
+
+#include "openvino/pass/graph_rewrite.hpp"
+
+namespace ov {
+namespace intel_cpu {
+
+class MarkUpRopeInputs: public ov::pass::MatcherPass {
+public:
+    OPENVINO_RTTI("MarkUpRopeInputs", "0");
+    MarkUpRopeInputs();
+private:
+    std::deque<std::shared_ptr<ov::Node>> nodes;
+    std::unordered_set<std::shared_ptr<ov::Node>> visited;
```

Comment:
Could you please explain why do we need these fields? Why can't we declare them in transformations callback directly?
I can imagine, why `visited` is needed (to avoid double markup of some nodes), but have no idea about `nodes`

Reply:
Hi the `nodes` is a `queue`, which is used for breadth first search. Because we use a `while` loop for `BFS`, a `queue` is needed for tracking the remaining candidates.

Reply:
remove the `nodes` from class declaration, move it to internal variable.